File: Program166.java
import java.lang.*;
import java.util.*;

class Bitwise
{
   public boolean CheckBit(int iNo)
   {
      
    int iMask = 4; 
    int iResult = 0;

    iResult = iNo & iMask;

    if(iResult == 0)
    {
        return false;
    }
    else
    {
        return true;
    }

   }

}

class Program166
{
    public static void main(String arg[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter number");
        int iValue = sobj.nextInt();
        
        Bitwise bobj = new Bitwise();
        boolean bRet = bobj.CheckBit(iValue);

        if(bRet == true)
        {
            System.out.println("Bit is on");
        }
        else
        {
            System.out.println("Bit is off");
        }

    }
}

File: Program166.c
#include<stdio.h>
#include<stdbool.h>

bool CheckBit(int iNo)
{
   int iMask = 4;
   int iResult = 0;

   iResult = iMask & iNo;

   if(iResult == 0)
   {
       return false;
   }
   else
   {
       return true;
   }
}

int main()
{
    int iValue = 0;
    bool bRet;

    printf("Enter number\n");
    scanf("%d",&iValue);

    bRet= CheckBit(iValue);

    if(bRet == true)
    {
        printf("Bit is on");
    }
    else
    {
        printf("Bit is off");
    }

    return 0;
}